src/test/regress/sql/shared_connection_stats.sql

```diff
@@ -190,6 +190,113 @@ BEGIN;
 		hostname, port;
 COMMIT;
 
+BEGIN;
```

Comment:
1. Maybe also add similar tests for append distributed, range distributed, and reference tables, and verify that our understanding of how they should behave is correct. To create range distributed table shards, you can use `create_range_partitioned_shards`.
2. Can you try to see if we can use `\COPY ... FROM PROGRAM 'seq 32'` to keep the test shorter?

Reply:
also insert..select tests that uses the COPY logic underneath (especially with RETURNING/ON CONFLICT, because the logic is a bit different).

Reply:
bump

Reply:
Added some more tests
- [x] `INSERT INTO SELECT ... RETURNING *`
- [x] range distribution
- [x] reference tables
- [ ] append distribution